src/database/operations/generators/generateItemLoader.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Generates an item loader function for a specific query and table.
+ *
+ * @template T - The type of the item being loaded.
+ * @template P - The type of the parameters for the query. Use labeled tuples so the generated function has a pretty type definition.
+ *
+ * @param {string} queryName - The name of the query to execute.
+ * @param {string} tableName - The name of the table to load metrics from.
+ * @param {Object} parameterNames - An object mapping parameter indices to their names within the query.
+ *
+ * @returns {Function} A function that takes query parameters, limit, and offset, and returns a promise that resolves to a bundle of entries and total count.
+ */
+
+import { db } from '../../db';
+import { NotFoundError } from '../../../errors';
+import type { JsonResultSet } from '../../../types';
+
+const generateItemLoader = <T, P extends [...args: unknown[]]>(
```

Comment:
I think that as we add verbs (create, update, etc) it will be cleaner for this to be called `generateLoadItemOperation` rather than `generateItemLoader`.

e.g. `generateCreateOrUpdateItemOperation` is a cleaner name than something like `generateItemCreateOrUpdater`

A little more verbose, but hash-tag-worth imho